update select: 去掉了componentWillReceiveProps

fondadam opened the pull request.

liyatang commented:
![image](https://user-images.githubusercontent.com/1010130/35499768-bd34e67a-050e-11e8-9436-45d7083d4d5b.png)

1 不应该用 map

2 就不用判断 Option 了，暂时不用做这么强的检测。 一般不会用错，谁用错怪谁。

3 建议 Object.assign 的 第二个和第三个参数可换下位置。

liyatang commented:
```
const findItemByValueFromList = (childrenList, val) => {
	let ret = null;
	React.Children.map(childrenList, (el) => {
		if(el.type.displayName === 'Option') {
			const elProps = el.props;
			const elValue = elProps.value;
			
			if(elValue === val) {
				ret = elProps.children;
			}
		}
	});
	return ret;
};
```

1 不应该用 map

2 参数 childrenList 是个列表，而这里应该是 children ，可能是个obj，可能是个 列表。 你 文档上的demo都是 _.map 的，所以传的必定是个列表。    可能是个obj的场景如下

```
<Select value={value} onChange={this.handleChange}>
    <Option value={1}>1</Option>
</Select>
```

JoeBBChen commented:
我看 `Select` 没有 name ，那如果我 `Select` 包在 Form 里，用 `onSubmit` 提交，没有 name ，那不是没了那个字段？

liyatang commented:
js
1 findItemChildByValFromChildren 里不用 each 等遍历方法。  findItem 是应该返回 Item, 而不是 Item.children
2 handleSelectionClick handleOptionClick 为什么要 return false ？
3 为什么要 onChange && onChange(xxx) onChange 是 isRequired
4 Option 有 disabled 这种用法？ Option 组件并没有说明。 但是 Select.js line:72 用了

css
1 ![](http://7xlnio.com1.z0.glb.clouddn.com/18-1-31/29727225.jpg)
2 三角用统一的 gm-arrow
3 hover bg色用变量 line:69
4 selected bg色用 line:82

疑问
1 gm-select-padding: 8px 12px 怎么来？

liyatang commented:
![](http://7xlnio.com1.z0.glb.clouddn.com/18-1-31/97007585.jpg)
option.js 中的 rest 包含了 value，然后给了 div 有点不合理。 div 并没有 value 这个属性

fondadam commented:
js:
1. 我返回的是item.children, 所以是findItemChild
2. 因为需要返回，当然，更好的是 return undefined 或者 直接return
3. fixed
4. Option 已添加 disabled 参数说明
5. fixed

css:
1. fixed （ 设置了最低高度
2. fixed
3. fixed
4. fixed
5. 我看了下，开了 ( 提交时不是也有stylelint吗？
![image](https://user-images.githubusercontent.com/7132998/35614336-ad7b6910-06a9-11e8-96ca-f44ca671a520.png)

疑问：
1. 参考bootstrap，设置padding为： 6px  12px

liyatang commented:
pass
